### PR TITLE
Integrate Kustomize

### PR DESCRIPTION
**Pull request by clebs**
**Description**

Changes proposed in this pull request:

- Kustomize integrated for rendering of kubebuilder operators
- Added first version of the deploy command that installs lifecycle and module managers via kustomize
- Added dry run mode: prints all manifests to stdout but does not apply them.

**Related issue(s)**
#1318

### CODE REVIEW

**Comment by Tomasz-Smelcerz-SAP** [comment]
While running against a Gardener GCP cluster, I got the following error:
```
i303773 √ ..ng/src/github.com/kyma-project/cli/ % go run cmd/main.go alpha deploy
- Kustomize ready
/ Deploying KymaError: could not apply operator for lifecycle-manager: failed to create typed patch object (kcp-system/lifecycle-manager-controller-resources-metrics-dashboard; /v1, Kind=ConfigMap): .metadata.labels.grafana_dashboard: expected string, got &value.valueUnstructured{Value:1}
exit status 1
```

**Comment by Tomasz-Smelcerz-SAP** [comment]
It also happens in a k3d cluster, and also when I am using a compiled binary. It looks like something's wrong with kustomize configuration

**Comment by clebs** [comment]
Hey @Tomasz-Smelcerz-SAP,
yes I am facing the same issue now in both k3d and gardener too. Please keep in mind that the deploy currently fetches whatever is in `main` for both managers. This is a breaking change to the operator templating caused by [this commit](https://github.com/kyma-project/lifecycle-manager/commit/a7b87038f9764b3d26ba6238fb73e6ac77d9e06f).

I will debug it with @ruanxin, but in the meantime if you want to test a working state, you can use the flag `--source/-s local` and set your local clone of the managers to a previous commit.

**Comment by Tomasz-Smelcerz-SAP** [comment]
It's probably related to [this commit](https://github.com/kyma-project/lifecycle-manager/commit/a7b87038f9764b3d26ba6238fb73e6ac77d9e06f) That introduced a label with an integer value.

File: internal/deploy/workspace.go [hunk 1]
```diff
@@ -119,6 +110,25 @@ func ResolveLocalWorkspacePath(ws string, local bool) (string, error) {
 	return ws, nil
 }
 
+// resolveLocalRepo tries to find the repository with the given name in the GOPATH
+// the repository must have its full name starting from $GOPATH/src
+func resolveLocalRepo(repo string) (string, error) {
```
Comment (Tomasz-Smelcerz-SAP) [inline internal/deploy/workspace.go:115]: I think there is a problem here. GOPATH should point to the workspace dir. In this directory there are "src" and "bin" subdirectories, and project sources should be located under "src" [(docs)](https://go.dev/doc/gopath_code#GOPATH)
On my system the "src" dir is not taken into account and I go this:
```
% echo $GOPATH
/Users/tsm/Documents/development/golang
% ./bin/kyma-darwin alpha deploy --source=local
- Kustomize ready
Error: error resolving lifecycle-manager: Local Kyma source directory '/Users/tsm/Documents/development/golang/github.com/kyma-project/lifecycle-manager' not found
% file $GOPATH/src/github.com/kyma-project/lifecycle-manager
/Users/tsm/Documents/development/golang/src/github.com/kyma-project/lifecycle-manager: directory
```

Comment (clebs) [reply internal/deploy/workspace.go:115]: Thanks for catching that! Yes indeed there is a bug there; the `src` path element is missing.
Fixed it.

**Review by Tomasz-Smelcerz-SAP — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Pull request merged**